erts/emulator/beam/erl_init.c

```diff
@@ -762,6 +806,110 @@ static void ethr_ll_free(void *ptr)
     erts_free(ERTS_ALC_T_ETHR_LL, ptr);
 }
 
+#if defined(__linux__) && defined(__x86_64__) && defined(__LP64__)
+static int
+is_transparent_hugepage_madvise_enabled(void)
```

Comment:
This feels a bit too platform-specific to have in `erl_init.c`, can you move the meat of it to `erts/emulator/sys/unix/sys_huge_pages.c` or similar?

Reply:
Sure, I can move that.  I am not sure what the best place would be but I'll make the suggested change.  If it doesn't look good or doesn't factor well, we can always move it again.